src/constants/MacroSettingsMap.ts

```diff
@@ -0,0 +1,36 @@
+import { SettingsCategory } from './enums'
+
+export interface MacroSettingInfo {
+  pageIndex: number
+  category: SettingsCategory
+  displayString: string
+}
+
+export class MacroSettingsGroup {
+  static get Notifications(): MacroSettingInfo {
+    return {
+      pageIndex: 0,
+      category: SettingsCategory.Macro,
+      displayString: 'Notifications'
+    }
+  }
+  static get SequenceDefaults(): MacroSettingInfo {
+    return {
+      pageIndex: 1,
+      category: SettingsCategory.Macro,
+      displayString: 'Sequence'
+    }
+  }
+
+
+  static readonly all: MacroSettingInfo[] = [
+    MacroSettingsGroup.Notifications,
+    MacroSettingsGroup.SequenceDefaults,
+  ]
+}
+
+export const macroSettingInfoLookup = new Map<number, MacroSettingInfo>(
+  MacroSettingsGroup.all
+    .filter((setting) => setting.pageIndex !== undefined)
+    .map((setting) => [setting.pageIndex!, setting])
+)
```

Comment:
Everything in this file is only used in the `MacroSettingsModal`, it seems kinda redundant to me to build this up like this. When you could just have a simple definition inside of the `MacroSettingsModal` file and even reference the components directly in the definition there as well, making it even more generic than it is now

Reply:
Something like this?: [452e414](https://github.com/WootingKb/wooting-macros/pull/194/commits/452e414d1dd538b773936c5dda25601eb9227112)

Reply:
Should be this: [e800afe](https://github.com/WootingKb/wooting-macros/pull/194/commits/e800afed29526bc34961a92b2f9945d23c4294f4)

and this:
[8208a10](https://github.com/WootingKb/wooting-macros/pull/194/commits/8208a1007ef71d935822a887e01ee61505b941d7)